playbook/app/pb_kits/playbook/pb_date_picker/plugins/quickPick.tsx

```diff
@@ -0,0 +1,185 @@
+    const ranges = {
+      'Today': [new Date(), new Date()],
+      'Yesterday': [moment().subtract(1, 'days').toDate(), moment().subtract(1, 'days').toDate()],
+      'This week': [moment().startOf('isoWeek').toDate(), thisWeekEndDate],
+      'This month': [moment().startOf('month').toDate(), thisMonthEndDate],
+      'This quarter': [moment().startOf('quarter').toDate(), thisQuarterEndDate],
+      'This year': [moment().startOf('year').toDate(), thisYearEndDate],
+      'Last week': [
+        moment().subtract(1, 'week').startOf('isoWeek').toDate(),
+        moment().subtract(1, 'week').endOf('isoWeek').toDate()
+      ],
+      'Last month': [
+        moment().subtract(1, 'month').startOf('month').toDate(),
+        moment().subtract(1, 'month').endOf('month').toDate()
+      ],
+      'Last quarter': [
+        moment().subtract(1, 'quarter').startOf('quarter').toDate(),
+        moment().subtract(1, 'quarter').endOf('quarter').toDate()
+      ],
+      'Last year': [
+        moment().subtract(1, 'year').startOf('year').toDate(),
+        moment().subtract(1, 'year').endOf('year').toDate()
+      ]
+    }
+    //creating the ul element for the nav dropdown and giving it classnames
+    const rangesNav = document.createElement('ul');
+
+    // creating the pluginData object that will hold the properties of this plugin
+    const pluginData: pluginDataType = {
+      ranges: ranges,
+      rangesNav: rangesNav,
+      rangesButtons: [],
+    };
+
+    /**
+   *    @param {string} label
+   *    @returns HTML Element
+   */
+
+    //function for creating the range buttons in the nav
+    const addRangeButton = (label: string) => {
+
+      // creating new elements to mimick selectable card component
+      const div2 = document.createElement('div');
+      div2.className = "nav-item-link"
+      div2.innerHTML = label;
+
+      pluginData.rangesButtons[label] = div2;
+
+      // create li elements inside the dropdown
+      const item = document.createElement('li');
+      item.className = "nav-item";
+
+      // append those nav items to the li items
+      item.appendChild(pluginData.rangesButtons[label]);
+
+      // append the li item to the ul rangeNav prop
+      pluginData.rangesNav.appendChild(item);
+
+      // return the ranges buton prop
+      return pluginData.rangesButtons[label];
+    };
+
+    const selectActiveRangeButton = (selectedDates: Array<Date>) => {
+      const current = pluginData.rangesNav.querySelector('.active');
+
+      if (current) {
+        current.classList.remove('active');
+      }
+      /** conditional statment to extract start and end dates from selectedDates,
+      *   then loop through ranges prop in pluginData
+      *   and check if chosen dates equal to a date in the ranges prop
+      *   if they are equal, add the active class
+      */
+     
+      if (selectedDates.length > 0 && activeLabel) {
+        // const selected = pluginData.rangesNav.querySelectorAll(".nav-item-link")
```

Comment:
same as above, if this isn't needed let's delete it instead of leaving it commented out!